lib/push.js

```diff
@@ -253,190 +253,148 @@ module.exports = function (log, db, config) {
+     *   @param {String[]} [options.excludedDeviceIds]
+     *   @param {String} [options.data]
+     *   @param {String} [options.TTL] (in seconds)
      * @promise
      */
-    notifyUpdate: function notifyUpdate(uid, reason) {
-      reason = reason || 'accountVerify'
-      return this.pushToAllDevices(uid, reason)
-    },
+    notifyUpdate (uid, devices, reason, options = {}) {
+      if (options.includedDeviceIds) {
+        const include = new Set(options.includedDeviceIds)
+        devices = devices.filter(device => include.has(device.id))
 
-    /**
-     * Notifies all devices (except the one who joined) that a new device joined the account
-     *
-     * @param {String} uid
-     * @param {String} deviceName
-     * @param {String} currentDeviceId
-     * @promise
-     */
-    notifyDeviceConnected: function notifyDeviceConnected(uid, deviceName, currentDeviceId) {
-      var data = Buffer.from(JSON.stringify({
-        version: PUSH_PAYLOAD_SCHEMA_VERSION,
-        command: PUSH_COMMANDS.DEVICE_CONNECTED,
-        data: {
-          deviceName: deviceName
+        if (devices.length === 0) {
+          return P.reject('devices empty')
         }
-      }))
-      var options = { data: data, excludedDeviceIds: [currentDeviceId] }
-      return this.pushToAllDevices(uid, 'deviceConnected', options)
-    },
+      } else if (options.excludedDeviceIds) {
+        const exclude = new Set(options.excludedDeviceIds)
+        devices = devices.filter(device => ! exclude.has(device.id))
+      }
 
-    /**
-     * Notifies a device that it is now disconnected from the account
-     *
-     * @param {String} uid
-     * @param {String} idToDisconnect
-     * @promise
-     */
-    notifyDeviceDisconnected: function notifyDeviceDisconnected(uid, idToDisconnect) {
-      var data = Buffer.from(JSON.stringify({
-        version: PUSH_PAYLOAD_SCHEMA_VERSION,
-        command: PUSH_COMMANDS.DEVICE_DISCONNECTED,
-        data: {
-          id: idToDisconnect
-        }
-      }))
-      var options = { data: data, TTL: TTL_DEVICE_DISCONNECTED }
-      return this.pushToAllDevices(uid, 'deviceDisconnected', options)
+      return this.sendPush(uid, devices, reason, filterOptions(options))
     },
 
     /**
-     * Notifies all devices that a the profile attached to the account was updated
+     * Notify devices (except currentDeviceId) that a new device was connected
      *
      * @param {String} uid
+     * @param {Device[]} devices
+     * @param {String} deviceName
+     * @param {String} currentDeviceId
      * @promise
      */
-    notifyProfileUpdated: function notifyProfileUpdated(uid) {
-      var data = Buffer.from(JSON.stringify({
-        version: PUSH_PAYLOAD_SCHEMA_VERSION,
-        command: PUSH_COMMANDS.PROFILE_UPDATED
-      }))
-      var options = { data: data }
-      return this.pushToAllDevices(uid, 'profileUpdated', options)
+    notifyDeviceConnected (uid, devices, deviceName, currentDeviceId) {
```

Comment:
It's made a pig's ear of the diff here, sorry. These methods should all be compared against themselves earlier on in the deleted lines.